File: AndroidJSONExample/app/src/main/java/com/example/shakil/androidjsonexample/MainActivity.java
package com.example.shakil.androidjsonexample;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import com.example.shakil.androidjsonexample.Adapter.MovieAdapter;
import com.example.shakil.androidjsonexample.Common.Common;
import com.example.shakil.androidjsonexample.Interface.RetrofitService;
import com.example.shakil.androidjsonexample.Model.Movie;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MainActivity extends AppCompatActivity {

    RecyclerView recyclerMovie;
    RetrofitService mService;
    RecyclerView.LayoutManager layoutManager;
    MovieAdapter adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mService = Common.getMovies();

        recyclerMovie = findViewById(R.id.recyclerMovie);
        recyclerMovie.setHasFixedSize(true);
        layoutManager = new LinearLayoutManager(this);
        recyclerMovie.setLayoutManager(layoutManager);

        getAllMovieList();

    }

    private void getAllMovieList() {
        mService.getAllMovies().enqueue(new Callback<List<Movie>>() {
            @Override
            public void onResponse(Call<List<Movie>> call, Response<List<Movie>> response) {
                adapter = new MovieAdapter(getBaseContext(), response.body());
                adapter.notifyDataSetChanged();
                recyclerMovie.setAdapter(adapter);
            }

            @Override
            public void onFailure(Call<List<Movie>> call, Throwable t) {

            }
        });
    }
}

File: KotlinJSONExample/app/src/main/java/com/example/shakil/kotlinjsonexample/MainActivity.kt
package com.example.shakil.kotlinjsonexample

import android.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.shakil.kotlinjsonexample.Adapter.MyMovieAdapter
import com.example.shakil.kotlinjsonexample.Common.Common
import com.example.shakil.kotlinjsonexample.Interface.RetrofitService
import com.example.shakil.kotlinjsonexample.Model.Movie
import dmax.dialog.SpotsDialog
import kotlinx.android.synthetic.main.activity_main.*
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MainActivity : AppCompatActivity() {

    lateinit var mService : RetrofitService
    lateinit var layoutManager: LinearLayoutManager
    lateinit var adapter : MyMovieAdapter
    lateinit var dialog: AlertDialog

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        mService = Common.retrofitService

        recyclerMovieList.setHasFixedSize(true)
        layoutManager = LinearLayoutManager(this)
        recyclerMovieList.layoutManager = layoutManager

        dialog = SpotsDialog.Builder().setCancelable(false).setContext(this).build()

        getAllMovieList()
    }

    private fun getAllMovieList() {
        dialog.show()

        mService.getMovieList().enqueue(object : Callback<MutableList<Movie>> {
            override fun onFailure(call: Call<MutableList<Movie>>, t: Throwable) {
                TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
            }

            override fun onResponse(call: Call<MutableList<Movie>>, response: Response<MutableList<Movie>>) {
                adapter = MyMovieAdapter(baseContext, response.body() as MutableList<Movie>)
                adapter.notifyDataSetChanged()
                recyclerMovieList.adapter = adapter

                dialog.dismiss()
            }

        })
    }
}
